Added 'config core' and 'show cores' CLI commands

rajendra-dendukuri opened the pull request:
Signed-off-by: Rajendra Dendukuri <rajendra.dendukuri@broadcom.com>

<!--
Please make sure you've read and understood our contributing guidelines:
https://github.com/Azure/SONiC/blob/gh-pages/CONTRIBUTING.md

** Make sure all your commits include a signature generated with `git commit -s` **

If this is a bug fix, make sure your description includes "closes #xxxx",
"fixes #xxxx" or "resolves #xxxx" so that GitHub automatically closes the related
issue when the PR is merged.

If you are adding/modifying/removing any command or utility script, please also
make sure to add/modify/remove any unit tests from the tests
directory as appropriate.

If you are modifying or removing an existing 'show', 'config' or 'sonic-clear'
subcommand, or you are adding a new subcommand, please make sure you also
update the Command Line Reference Guide (doc/Command-Reference.md) to reflect
your changes.

Please provide the following information:
-->

**- What I did**
Added wrapper CLI commands to see core files information. Also added CLI command to enable/disable
corefile generation.

**- How I did it**
Added new commands
config core
show cores

**- How to verify it**
show cores list
config core enable
config core disable

**- Previous command output (if the output of a command-line utility has changed)**

**- New command output (if the output of a command-line utility has changed)**
root@sonic:/home/admin# show cores
Usage: show cores [OPTIONS] COMMAND [ARGS]...

Show core dump events encountered

Options:
-?, -h, --help Show this message and exit.

Commands:
config Show coredump configuration
info Show information about one or more coredumps
list List available coredumps
root@sonic:/home/admin# show cores list
TIME PID UID GID SIG COREFILE EXE
Sat 2016-12-31 22:46:22 UTC 4933 0 0 6 present /usr/bin/vlanmgrd
root@sonic:/home/admin# show cores info
PID: 4933 (vlanmgrd)
UID: 0 (root)
GID: 0 (root)
Signal: 6 (ABRT)
Timestamp: Sat 2016-12-31 22:46:22 UTC (6min ago)
Command Line: /usr/bin/vlanmgrd
Executable: /usr/bin/vlanmgrd
Control Group: /docker/38e40a3313bd46ff4e282f897cdc25502a2b15febe9839715f92b035b6f2cffe
Slice: -.slice
Boot ID: 3f776b83fa704d089bd81e0b108c2a09
Machine ID: a0c0523aae8c4c21a8d9fa12e18d85b0
Hostname: sonic
Storage: /var/lib/systemd/coredump/core.vlanmgrd.0.3f776b83fa704d089bd81e0b108c2a09.4933.1483224382000000000000.lz4
Message: Process 4933 (vlanmgrd) of user 0 dumped core.

jleveque reviewed (changes requested):
We are in the process of splitting groups out into their own files and importing them in the main.py files, for better organization and smaller files. Can you please move these new groups and their commands into show/cores.py and config/cores.py? (I also suggest aligning the groups to both be plural).

rajendra-dendukuri commented:
> We are in the process of splitting groups out into their own files and importing them in the main.py files, for better organization and smaller files. Can you please move these new groups and their commands into show/cores.py and config/cores.py? (I also suggest aligning the groups to both be plural).

Renamed the commands to "coredump" to be inline with kdump. Also split them out into separate files config/coredump.py and show/coredump.py.

rajendra-dendukuri commented:
This PR requires https://github.com/Azure/sonic-buildimage/pull/6079
